### PR TITLE
Stats details

### PR DESCRIPTION
**Pull request by peter-budo**
@karllindmark 

Build http://ci.peterscorner.co.uk/job/BF4-Intel-PR/15/

Details stats, for the moment just plain list with headers and list of items, that we can later style it easily

### CODE REVIEW

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/model/stats/details/StatsDetailsGrouped.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        detailsList.add(new DetailsListItem(R.string.highest_kill_streak,  stringValueOf(details.getHighestKillStreak())));
+        detailsList.add(new DetailsListItem(R.string.nemesis_kills,  stringValueOf(details.getNemesisKills())));
+        detailsList.add(new DetailsListItem(R.string.highest_nemesis_streak,  stringValueOf(details.getHighestNemesisStreak())));
+
+        detailsList.add(new BaseListHeader(R.string.game_mode_extra));
+        detailsList.add(new DetailsListItem(R.string.flags_captured,  stringValueOf(details.getFlagsCaptured())));
+        detailsList.add(new DetailsListItem(R.string.flags_defended,  stringValueOf(details.getFlagsDefended())));
+    }
+
+    private String stringValueOf(int value) {
```
Comment (karllindmark) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/model/stats/details/StatsDetailsGrouped.java:88]: We should consider extracting the functionality of the `stringValueOf(...)` to some helper/utility class, as we may use the same functionality elsewhere? :cake:

File: core/src/test/java/com/ninetwozero/bf4intel/json/stats/details/StatsDetailsTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertNotNull;
+
+public class StatsDetailsTest {
+
+    @Test
+    public void should_successfully_parse_json() throws IOException {
+        StatsDetails details = IntelJsonParser.parse("/json/details-stats.json", StatsDetails.class);
+        assertNotNull(details);
```
Comment (karllindmark) [inline core/src/test/java/com/ninetwozero/bf4intel/json/stats/details/StatsDetailsTest.java]: Not that I'm a test guru or anything, but I'd `assertNotNull()` on a getter in `StatsDetails`, as `StatsDetails` will often be an actual (non null) instance, but with correct/incorrect (null) values depending on the parsed resource. If I'm not mistaken that is. :-)

File: BF4Intel/src/main/java/com/ninetwozero/bf4intel/model/stats/details/StatsDetailsGrouped.java (hunk 1 above)
Comment (peter-budo) [inline BF4Intel/src/main/java/com/ninetwozero/bf4intel/model/stats/details/StatsDetailsGrouped.java:88]: When the need arises (example me renaming BattleReportHeader to BaseHeader). So at the moment used in only one place therefore no need.

File: core/src/test/java/com/ninetwozero/bf4intel/json/stats/details/StatsDetailsTest.java (hunk 2 above)
Comment (peter-budo) [inline core/src/test/java/com/ninetwozero/bf4intel/json/stats/details/StatsDetailsTest.java]: Yeah forgot to change it once I realised that parsing is not getting me data from parsed JSON LOL

Comment (karllindmark) [inline core/src/test/java/com/ninetwozero/bf4intel/json/stats/details/StatsDetailsTest.java]: :+1:

**Comment by karllindmark** [comment]
I'll try this out later on-device as well!

**Comment by peter-budo** [comment]
@karllindmark test updated

**Comment by karllindmark** [comment]
OK, a few comments now that I've run the app - mostly regarding the layout: 
- [x] Score probably needs correct formatting (ie, 1000000 => 1,000,000). This would make it easier to "understand" how much a value is.
- [ ] Text in the value column should probably be aligned from the right (ie, the gravity seems off)
- [ ]  "Extra" & "Game Mode Extra" - any "more interesting" name we can use for those two sections?- [ ] - [ ] White or grey (current) background? Is grey still suitable even when a no card is available?

Any thoughts on my points above? :lantern: 

@peter-budo

**Comment by peter-budo** [comment]
- Agree on number formatting
- Disagree, you need text to start from left, it makes reading easier then zig-zag of words starting there. It would be much better to do every second item in different background colour to lead the eyes
- We should not be introducing new "vocabulary" when user know these already exists. It is different question if we want to show other stats the are available but not present in web presentation

**Comment by peter-budo** [comment]
@karllindmark number formatting done and also added some spacing to detail list items

**Comment by karllindmark** [comment]
Right column should probably be same width as the widest item, but left-aligned. Or am I lost? :l

**Comment by peter-budo** [comment]
It is linear layout. Wouldn't that make it zig-zag? I see no issues with current layout

**Comment by karllindmark** [comment]
Not necessarily - would it be possible to do it as `TableLayout`/`GridLayout` instead to make sure that we have consistent widths? What looks weird to me is the somewhat centered (and left-aligned) column (ie, the actual value column). I'm usually a right-align guy, but I asked around and learned that left-aligning is more natural in a situation like that. 

To sum it up - the spacing to the right looks a bit too much - is there any more discrete variant we could do? Just pitching thoughts - other than that it works as expected to.

**Comment by peter-budo** [comment]
Sorry this is too long discussion and should be taken on Skype because we are arguing about trivial thing. As for the layouts GridLayout it is overkill, because we want to show simple list of name and the value, sometimes show some header. So nothing nowhere close to assignments or awards. TableLayout is just horrid to use as you can see from my previous commits.

Spacing to the right of what, label or the value? Label is padded from left to not sit directly on view border. There is no spacing on value. Layout has weight distribution of 0.65 label and 0.35 value

**Comment by karllindmark** [comment]
Correct Mr Budo - I'll merge the PR and we can look into UI later on. :+1:

**Pull request merged**